meta/src/optimizer/restorer.rs

```diff
@@ -133,33 +127,37 @@ mod tests {
             )))))))
         };
 
-        assert_eq!(restore_on_err(rules[0].clone(), &to_hash_map(&rules)), restored);
+        assert_eq!(
+            restore_on_err(rules[0].clone(), &to_hash_map(&rules)),
+            restored
+        );
     }
 
     #[test]
     fn restore_choice_branch_with_and_branch_without() {
-        let rules = vec![
-            OptimizedRule {
-                name: "rule".to_owned(),
-                ty: RuleType::Normal,
-                expr: Choice(
-                    Box::new(Push(Box::new(Str(String::from("a"))))),
-                    Box::new(Str(String::from("a")))
-                )
-            },
-        ];
+        let rules = vec![OptimizedRule {
+            name: "rule".to_owned(),
+            ty: RuleType::Normal,
+            expr: Choice(
+                Box::new(Push(Box::new(Str(String::from("a"))))),
+                Box::new(Str(String::from("a")))
+            )
+        }];
 
         let restored = OptimizedRule {
             name: "rule".to_owned(),
             ty: RuleType::Normal,
             expr: Choice(
                 Box::new(RestoreOnErr(Box::new(Push(Box::new(Str(String::from(
-                    "a"
```

Comment:
What an eye-sore. Maybe this should be store in a variable just to make things less ugly?

Reply:
Honestly, it might be worth it to make a very-simple-to-translate local macro_rules to write these test cases in. Honestly, just something to box up everything would be helpful.

I made a limited version of that in 72dc062e793d98c1b3a123d348bd68908fbcf8ec to show it off, this is very much a your call thing though.

Reply:
Sure, definitely! In pursuit of thoroughness, I often lose track of code quality within tests.

Reply:
Any update on this?

Reply:
Yes, the [macro exists](https://github.com/pest-parser/pest/pull/315/files#diff-daba655efcad6f906c92c455f29e0e14R14) and is used elsewhere in the optimizer, I just missed this file 😅

I've realized just now that other crates might also want a `box_tree!` for their tests, namely `generator/generator.rs` at least. I can add that in a later PR, I'd like to get this one merged as soon as realistic to avoid growing stale since it (necessarily) touches so much, tbh.